### ISSUE TITLE
system task pool

### ISSUE THREAD

**Issue by chegewara** [aws/amazon-freertos#793]
Hi,
im not sure this is bug, but here is my story.
For last few days i am trying to recover as much memory as i can from all tasks. One place i thought i can recover some heap is iot_thread tasks (or pool of tasks). There is 3 tasks with name `iot_thread`. One is created by demo_runner and other 2 are created by pool tasks.
I have task where i can check stack water mark and tasks created by pool task have very high value (4500-5500). Ive been trying to lower tasks stack, but this is causing hang or crash in app.
I also wanted to change this line to create only one task instead of two:
https://github.com/aws/amazon-freertos/blob/release-1.5/libraries/c_sdk/standard/common/iot_init.c#L75
so i tried with `IOT_TASKPOOL_INFO_INITIALIZER_SMALL`. You can imagine my surprise when i still had the same amount of `iot_thread` tasks.

Long story short, when i decrease `IOT_THREAD_DEFAULT_STACK_SIZE` then app crash/hang most the time and changing tasks pool with `IOT_TASKPOOL_INFO_INITIALIZER_SMALL` to only one does nothing.

Any help and advice is welcome. Thanks.

PS i am working with branch release-1.5 on esp32

**Comment by chegewara** [comment]
Logs with stack high water mark values:
```
I (15168) _tasksStatus: [0]Task stackInfo => stack remaining: 2040
I (15178) _tasksStatus: [1]Task IDLE => stack remaining: 368
I (15178) _tasksStatus: [2]Task ledsT => stack remaining: 892
I (15188) _tasksStatus: [3]Task iot_thread => stack remaining: 2496 <--- demo runner
I (15198) _tasksStatus: [4]Task NetRecv => stack remaining: 1212
I (15198) _tasksStatus: [5]Task IP-task => stack remaining: 1896
I (15208) _tasksStatus: [6]Task iot_thread => stack remaining: 5212 <--- pool tasks
I (15218) _tasksStatus: [7]Task iot_thread => stack remaining: 5200 <--- pool tasks
I (15228) _tasksStatus: [8]Task Tmr Svc => stack remaining: 3292
I (15228) _tasksStatus: [9]Task button_task => stack remaining: 820
I (15238) _tasksStatus: [10]Task MainShadowTask => stack remaining: 3308
I (15248) _tasksStatus: [11]Task btn_handler => stack remaining: 1752
I (15248) _tasksStatus: [12]Task eventTask => stack remaining: 904
I (15258) _tasksStatus: [13]Task WiFiConnectTask => stack remaining: 580
I (15268) _tasksStatus: [14]Task devStateT => stack remaining: 272
I (15268) _tasksStatus: [15]Task OTA Task => stack remaining: 2068
I (15278) _tasksStatus: [16]Task esp_timer => stack remaining: 3516
I (15288) _tasksStatus: [17]Task wifi => stack remaining: 1388
I (15298) _tasksStatus: [18]Task Logging => stack remaining: 1540
I (15298) _tasksStatus: [19]Task ipc0 => stack remaining: 728
I (15308) _tasksStatus: [20]Task mtr_handler => stack remaining: 3792
```
This is first time i didnt reset esp32 for some time and now i see that i cant change `iot_thread` task stack. So my question is why `IOT_TASKPOOL_INFO_INITIALIZER_SMALL` is not working:
```
I (835508) _tasksStatus: [1]Task IDLE => stack remaining: 368
I (835508) _tasksStatus: [2]Task IP-task => stack remaining: 1896
I (835518) _tasksStatus: [3]Task iot_thread => stack remaining: 2272 
I (835528) _tasksStatus: [4]Task ledsT => stack remaining: 892
I (835528) _tasksStatus: [5]Task NetRecv => stack remaining: 956
I (835538) _tasksStatus: [6]Task iot_thread => stack remaining: 872 <--- pool tasks
I (835548) _tasksStatus: [7]Task ShLSUpdt => stack remaining: 820
I (835548) _tasksStatus: [8]Task LS-Shd-0 => stack remaining: 1856
I (835558) _tasksStatus: [9]Task iot_thread => stack remaining: 5692 <--- pool tasks
I (835568) _tasksStatus: [10]Task iot_thread => stack remaining: 876 <--- pool tasks
I (835568) _tasksStatus: [11]Task Tmr Svc => stack remaining: 3292
I (835578) _tasksStatus: [12]Task button_task => stack remaining: 788
I (835588) _tasksStatus: [13]Task ipc0 => stack remaining: 728
I (835588) _tasksStatus: [14]Task mtr_handler => stack remaining: 3792
I (835598) _tasksStatus: [15]Task OTA Task => stack remaining: 2068
I (835608) _tasksStatus: [16]Task btn_handler => stack remaining: 1752
I (835618) _tasksStatus: [17]Task eventTask => stack remaining: 900
I (835618) _tasksStatus: [18]Task esp_timer => stack remaining: 3484
I (835638) _tasksStatus: [19]Task wifi => stack remaining: 1384
I (835638) _tasksStatus: [20]Task Logging => stack remaining: 1540
I (835648) _tasksStatus: [21]Task WiFiConnectTask => stack remaining: 580
I (835658) _tasksStatus: [22]Task devStateT => stack remaining: 272
```
